src/main/java/adris/altoclef/tasks/misc/GetToOuterEndIslandsTask.java

```diff
@@ -0,0 +1,120 @@
+package adris.altoclef.tasks.misc;
+
+import adris.altoclef.AltoClef;
+import adris.altoclef.Debug;
+import adris.altoclef.tasks.InteractWithBlockTask;
+import adris.altoclef.tasks.movement.GetToYTask;
+import adris.altoclef.tasks.resources.GetBuildingMaterialsTask;
+import adris.altoclef.tasks.speedrun.BeatMinecraft2Task;
+import adris.altoclef.tasks.squashed.CataloguedResourceTask;
+import adris.altoclef.tasksystem.Task;
+import adris.altoclef.util.Dimension;
+import adris.altoclef.util.ItemTarget;
+import adris.altoclef.util.baritone.GoalAnd;
+import adris.altoclef.util.helpers.StorageHelper;
+import adris.altoclef.util.helpers.WorldHelper;
+
+import baritone.api.pathing.goals.GoalComposite;
+import baritone.api.pathing.goals.GoalGetToBlock;
+import baritone.api.pathing.goals.GoalYLevel;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.Vec3i;
+
+public class GetToOuterEndIslandsTask extends Task {
+    private Task _beatTheGame;
+    private boolean _pathingToGateway = false;
+    public final Vec3i[] OFFSETS = {
+            new Vec3i(1, -1, 1),
+            new Vec3i(1, -1, -1),
+            new Vec3i(-1, -1, 1),
+            new Vec3i(-1, -1, -1),
+            new Vec3i(2, -1, 0),
+            new Vec3i(0, -1, 2),
+            new Vec3i(-2, -1, 0),
+            new Vec3i(0, -1, -2)
+    };
+    public GetToOuterEndIslandsTask() {
+
+    }
+
+    @Override
+    protected void onStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.END_GATEWAY);
+        _beatTheGame = new BeatMinecraft2Task();
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        if (isFinished(mod)) {
+            setDebugState("What?");
+            return null;
+        }
+        if (mod.getBlockTracker().anyFound(Blocks.END_GATEWAY)) {
+            if (!mod.getItemStorage().hasItemInventoryOnly(Items.ENDER_PEARL)) {
+                setDebugState("Getting an ender pearl");
+                return new CataloguedResourceTask(new ItemTarget(Items.ENDER_PEARL, 1));
+            }
+            BlockPos gateway = mod.getBlockTracker().getNearestTracking(Blocks.END_GATEWAY).get();
+            int blocksNeeded = Math.abs(mod.getPlayer().getBlockY() - gateway.getY()) +
```

Comment:
This was smart, although you should get at least 15 more blocks than needed. Experience has taught me that altoclef sometimes uses more blocks than it really needs. In these cases, it is good to have a backup.

Reply:
I would change this

Reply:
Why would it? Works perfectly fine for me. In fact, its using way less because of this sick new invention called walking on the ground so I feel like I'm already doing a backup

Reply:
I guess the problem is what if the bot has pillared up but it doesn't have enough blocks or something, it will try to mine the pillar. If half way down the pillar the bot decides "we have enough!" that's a problem since it will get stuck looping. We want the bot to finish mining the pillar and gather some materials from the ground, and that's why we usually ask for extra blocks.

If you for sure don't need that and know it won't get stuck (even in edge cases) then that's fine lol

Reply:
No it won't, because the amount of building materials it wants updates every time the bot changes block position so if mining the pillar moves it away from the gateway, it will still try to get more